.github/workflows/gradle.yml

```diff
@@ -855,158 +863,207 @@ jobs:
+
+      - name: Cache Build Artifacts
+        uses: ./.github/actions/cache-build-artifacts
+        with:
+          cache-key: ${{ secrets.CACHE_VERSION }}
+          cache-python: "false"
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9' 
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: "17"
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "lts/gallium"
+
+      - name: Install unzip
+        shell: bash
+        run: sudo apt-get update && sudo apt-get install -y unzip
+
+      - uses: azure/setup-helm@v3
+        with:
+          version: 'latest'
+          token: ${{ secrets.GITHUB_TOKEN }}
+        id: install
+
+      - uses: azure/setup-kubectl@v3
+        with:
+          version: 'v1.21.2' # default is latest stable
+
+      - name: Install tmate and it's dependencies
+        if: inputs.debug_mode == 'true'
+        shell: bash
+        run: |
+          sudo rm /var/lib/dpkg/lock
+          sudo rm /var/lib/dpkg/lock-frontend
+          sudo apt-get -y install tmate
+          sudo rm -f /var/lib/dpkg/lock
+          sudo rm -f /var/lib/dpkg/lock-frontend
+      
+      - name: Start tmate session in background
+        if: inputs.debug_mode == 'true'
+        shell: bash
+        run: |  
+          tmate -S /tmp/tmate.sock new-session -d               # Launch tmate in a headless mode
+          tmate -S /tmp/tmate.sock wait tmate-ready             # Blocks until the SSH connection is established
+          tmate -S /tmp/tmate.sock display -p '#{tmate_ssh}'    # Prints the SSH connection string
+          tmate -S /tmp/tmate.sock display -p '#{tmate_ssh_ro}' # Prints the read-only SSH connection string
+          tmate -S /tmp/tmate.sock display -p '#{tmate_web}'    # Prints the web connection string
+          tmate -S /tmp/tmate.sock display -p '#{tmate_web_ro}' # Prints the read-only web connection string
+
+      - name: Start a local k8s cluster
+        uses: jupyterhub/action-k3s-helm@v3
+        with:
+          k3s-version: v1.21.2+k3s1
+          helm-version: v3.9.4
+          extra-setup-args: --docker --kube-apiserver-arg service-node-port-range=6000-32767
+
+      - name: Try the cluster !
+        run: kubectl get pods -A
+
+      - name: Set up CI Gradle Properties
+        run: |
+          mkdir -p ~/.gradle/
+          cat > ~/.gradle/gradle.properties <<EOF
+          org.gradle.jvmargs=-Xmx8g -Xss4m --add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
+            --add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
+            --add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED \
+            --add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
+            --add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED
+          org.gradle.workers.max=512
+          org.gradle.vfs.watch=false
+          org.gradle.parallel=true
```

Comment:
shouldn't need this since it's in gradle.properties already

Reply:
Done